staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -350,8 +359,18 @@ var newETCD3Client = func(c storagebackend.TransportConfig) (*clientv3.Client, e
 		TLS:                  tlsConfig,
 		Logger:               etcd3ClientLogger,
 	}
-
-	return clientv3.New(cfg)
+	etcdClient, err := clientv3.New(cfg)
+	if err != nil {
+		return nil, err
+	}
+	consistentListFromCacheEnabled := utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache)
+	if consistentListFromCacheEnabled {
+		if err := checkVersion(etcdClient, dialTimeout); err != nil {
+			// Just warn to the user nothing more right now
+			klog.V(4).Infof("%v", err)
```

Comment:
We should disable the feature here. Please read https://github.com/kubernetes/kubernetes/issues/123192#issue-2125259649

> etcd version could not be acquired or etcd new enough - warn, but continue with whatever the current value of the feature gate is
> 
> etcd too old, feature gate not explicitly specified - warn and disable the feature
> etcd too old, feature gate explicitly enabled - abort
> etcd new enough - use whatever the current value of the feature gate is

Reply:
Thanks for comment, should we use something like this?
Can this disable the feature flag?
`utilfeature.DefaultMutableFeatureGate.OverrideDefault(features.ConsistentListFromCache, false)`

Reply:
I really appreciate comments. the code is updated and add the `utilfeature.DefaultMutableFeatureGate.OverrideDefault(features.ConsistentListFromCache, false)` for disabling feature, maybe we can test if that will work or not.

Reply:
Not sure.
cc @deads2k who might know more about this.